stdlib/private/StdlibUnittest/StdlibCoreExtras.swift

```diff
@@ -294,3 +294,21 @@ public struct LinearCongruentialGenerator: RandomNumberGenerator {
     return _state
   }
 }
+
+#if !SWIFT_ENABLE_REFLECTION
+
+extension String {
+  public init<T>(reflecting: T) {
+    self = "(reflection not available)"
+  }
+
+  public init<T>(describing: T) {
+    self = "(reflection not available)"
+  }
+}
+
+public func dump<T, TargetStream: TextOutputStream>(_ value: T, to target: inout TargetStream) {
+  target.write("(reflection not available)")
+}
+
+#endif
```

Comment:
The patch makes sense to me, but I was a bit disheartened by the sheer number of `#if SWIFT_ENABLE_REFLECTION` directives needed (77 files touched).

Would it be possible to leave Mirror declarations and uses intact, but "compile out" its implementation and return dummy values or crash.  Similarly to what we do here for String.